libraries/AP_Scripting/examples/fw_vtol_failsafe.lua

```diff
@@ -0,0 +1,140 @@
+-- This script allows for best effort return for quadplanes
+-- on loss of forward motor
+
+-- When an engine failure is detected (RPM and Vibration values drop below a certain threshold), the aircraft will:
+-- Send a warning ground station
+-- Start prioritizing airspeed completely
+-- Glide back towards the home location or rally point, monitoring the distance away from the point as well as Altitude.
+-- If the aircraft drops below a predetermined minimum altitude, QLAND mode is engaged and the aircraft lands at its current position.
+-- If the aircraft arrives within Q_FW_LND_APR_RAD of the return point before dropping below the minimum altitude, it should loiter down to the minimum altitude before switching to QRTL and landing.
+
+-- consider engine stopped when vibe is low and RPM low for more than 4s
+local ENGINE_STOPPED_MS = 4000
+
+-- RPM threshold below which engine may be stopped
+local RPM_LOW_THRESH = 500
+
+-- vibration threshold below which engine may be stopped
+local VIBE_LOW_THRESH = 2
+
+-- altitude threshold for QLAND
+local LOW_ALT_THRESH = 70
+
+-- when below MIN_AIRSPEED assume we are not in forward flight
+local MIN_AIRSPEED = 5
+
+-- time when engine stopped
+local engine_stop_ms = -1
+local engine_stopped = false
+
+-- have we triggered failsafe? Only trigger once
+local triggered_failsafe = false
+
+-- flight mode numbers for plane
+local MODE_AUTO = 10
+local MODE_RTL = 11
+local MODE_QLAND = 20
+
+-- update engine running status
+function check_engine()
+  local rpm = RPM:get_rpm(0)
+  if not rpm then
+     rpm = -1
+  end
+  local vibe = ahrs:get_vibration():length()
+
+  if not arming:is_armed() then
+     engine_stopped = false
+     engine_stop_ms = -1
+     return true
+  end
+
+  -- if either RPM is high or vibe is high then assume engine is running
+  if (rpm > RPM_LOW_THRESH) or (vibe > VIBE_LOW_THRESH) then
```

Comment:
```suggestion
  if (rpm and (rpm > RPM_LOW_THRESH)) or (vibe > VIBE_LOW_THRESH) then
```

This would let you remove the setting of rpm to -1 on line 41.

Reply:
done